lib/models/bundle.js

```diff
@@ -234,9 +44,9 @@ Bundle.prototype.validate = function () {
 
   codes.forEach(function (code) {
     // Single number refs -> real code.
-    if (code.ref !== null) {
-      var singleRef = codes[code.ref];
-      if (!singleRef.isCode) {
+    if (code.isSingleRef()) {
+      var singleRef = _.find({ id: code.singleRef })(codes);
```

Comment:
With object expressions, it's no longer safe to rely on the array indices of `codes`. We could consider converting the `codes` array into a hash of id keys/code pairs if we want O(1) lookups again.